### PR TITLE
Move load_sim_params to workflow

### PR DESCRIPTION
**Pull request by lispandfound**
Required for ucgmsim/qcore#309. Moves the `load_sim_params` function to workflow so that we can remove it from qcore.

### CODE REVIEW

**Review by claudio525 — APPROVED** [review]

**Review by sungeunbae — APPROVED** [review]

**Review by joelridden — COMMENTED** [review]
Only thing I'd say is to maybe have the sim_params.py in the automation folder of workflow

File: workflow/sim_params.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    load_fault: Union[bool, str] = True,
+    load_root: Union[bool, str] = True,
+    load_vm: Union[bool, str] = True,
+) -> dict:
+    """Load all necessary parameters for a simulation.
+
+    Parameters
+    ----------
+    sim_yaml_path : Union[bool, str]
+        Path to sim_params.yaml or a falsy value to not load it.
```
Comment (AndrewRidden-Harper) [inline workflow/sim_params.py:56]: I think this should be falsy --> False ?

File: workflow/sim_params.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    load_fault : Union[bool, str]
+        Either True, the path to fault_params.yaml or a falsy value to not load it.
```
Comment (AndrewRidden-Harper) [inline workflow/sim_params.py:58]: falsy --> false ?

**Review by AndrewRidden-Harper — APPROVED** [review]

File: workflow/sim_params.py (hunk 1 above)
Comment (lispandfound) [reply workflow/sim_params.py:56]: No that's a [well-defined term](https://stackoverflow.com/questions/39983695/what-is-truthy-and-falsy-how-is-it-different-from-true-and-false).

File: workflow/automation/submit/submit_sim_imcalc.py [hunk 3]
```diff
@@ -38,7 +35,7 @@ def submit_im_calc_slurm(
     two dictionaries, the passed in one has higher priority.
     """
     # Load the yaml params
-    params = utils.load_sim_params(
+    params = sim_params.load_sim_params(
```
Comment (joelridden) [inline workflow/automation/submit/submit_sim_imcalc.py:38]: missing import

**Review by joelridden — APPROVED** [review]

**Review by joelridden — APPROVED** [review]

**Review by AndrewRidden-Harper — APPROVED** [review]

**Review by claudio525 — APPROVED** [review]

**Review by sungeunbae — APPROVED** [review]

**Pull request merged**